k2/csrc/pytorch_context.cu

```diff
@@ -20,15 +20,22 @@
 
 namespace k2 {
 
+static std::once_flag has_cuda_init_flag;
+static bool has_cuda = false;
+static void InitHasCuda() {
```

Comment:
It may be much more compact if we can put `has_cuda` and `has_cuda_init_flag` into `(Init)HasCuda` (and return bool, i.e. static `has_cuda`, in this function). Then we can just do `if (HasCuda())`. And it's better to add `inline` here?

Reply:
`HasCuda()` involves an extra function call. `inline` is just a suggestion to the compiler.
I remeber that we added `K2_LOG` inside `CudaStreamOverride::Push` and the function
was no longer inlined.